doc/manual/contracts.md

```diff
@@ -718,6 +729,36 @@ value and continues with the second argument (here, our wrapped `value`).
 Let us see if we indeed preserved laziness:
 
 ```nickel #repl
+#hide-range{1-29}
+
+> let NumberBoolDict = fun label value =>
+    if std.is_record value then
+      let check_fields =
+        value
+        |> std.record.fields
+        |> std.array.fold_left
+          (
+            fun acc field_name =>
+              if std.string.is_match "^\\d+$" field_name then
+                acc # unused and always null through iteration
+              else
+                std.contract.blame_with_message "field name `%{field_name}` is not a number" label
+          )
+          null
+      in
+      value
+      |> std.record.map
+        (
+          fun name value =>
+            let label_with_msg =
+              std.contract.label.with_message "field `%{name}` is not a boolean" label
+            in
+            std.contract.apply Bool label_with_msg value
+        )
+      |> std.seq check_fields
+    else
+      std.contract.blame_with_message "not a record" label
+
```

Comment:
Can we have the repls chain together (as if they're one repl) rather than repeating these preambles everywhere?

Reply:
I think, this is actually the only place in the manual where this would have been useful. It would probably be feasible to add a REPL session id to the code block info string, but I'm not sure it's worth the effort right now.

Reply:
Yeah, fair enough.